content/typescript/concepts/utility-types/utility-types.md

```diff
@@ -0,0 +1,224 @@
+---
+Title: 'Utility Types'
+Description: 'Helps in working with data types more efficiently.'
+Subjects:
+  - 'Web Development'
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Utility Types'
+  - 'Data Types'
+  - 'TypeScript'
+CatalogContent:
+  - 'learn-typescript'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+TypeScript consists of several predefined generic types known as Utility Types. These utility types are useful for manipulating or creating new types more efficiently.
+
+## Partial
+
+Partial is used to create a type with all properties marked as optional.
+
+### Syntax
+
+A partial is written as:
+
+```pseudo
+Partial<T>;
+```
+
+Where `T` is the typo you want to make all the properties optional.
+
+### Partial Example
+
+In this example, all the properties of the `User` interface are optional:
+
+```ts
+interface User {
+  name: string;
+  age: number;
+  email: string;
+}
+
+const partialUser: Partial<User> = {
+  name: 'John',
+};
+```
+
+## Required
+
+`Required` is used to create a type with all properties of `T` marked as `Required`. It is the opposite of `Partial`.
+
+### Syntax
+
+`Required` in Typescript is written as:
+
+```pseudo
+Required<T>;
+```
+
+Where `T` is the typo you want to make all the properties mandatory.
+
+### Required Example
+
+In this example, all the properties of the `User` interface (including the optional properties) are made mandatory:
+
+```ts
+interface User {
+  name: string;
+  age?: number;
+}
+
+const requiredUser: Required<User> = {
```

Comment:
```suggestion
const requiredUser: Required<{ name: string; age: number}> = {
```
When we use Required utility type we must make sure to provide the values for all the properties that we have defined.